test/functional/AbstractValidatorBaseTest.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace Dhii\Validation\FuncTest;
+
+use Xpmock\TestCase;
+
+/**
+ * Tests {@see \Dhii\Validation\AbstractValidatorBase}.
+ *
+ * @since 0.1
+ */
+class AbstractValidatorBaseTest extends TestCase
+{
+    /**
+     * The class name of the test subject.
+     *
+     * @since 0.1
+     */
+    const TEST_SUBJECT_CLASSNAME = 'Dhii\\Validation\\AbstractValidatorBase';
+
+    /**
+     * Creates a new instance of the test subject.
+     *
+     * It will validate any boolean true value; all other values are invalid.
+     *
+     * @since 0.1
+     *
+     * @return AbstractValidator
+     */
+    public function createInstance()
+    {
+        $me = $this;
+        $mock = $this->mock(static::TEST_SUBJECT_CLASSNAME)
+                ->_getValidationErrors(function ($subject) {
+                    if ($subject !== true) {
+                        return array('Subject must be a boolean `true` value');
+                    }
+
+                    return array();
+                })
+                ->new();
+
+        return $mock;
+    }
+
+    /**
+     * Tests whether a valid instance of the test subject can be created.
+     *
+     * @since 0.1
+     */
+    public function testCanBeCreated()
+    {
+        $subject = $this->createInstance();
+
+        $this->assertInstanceOf(static::TEST_SUBJECT_CLASSNAME, $subject, 'Could not create a valid instance');
+    }
+
+    /**
+     * Tests whether valid values pass validation.
+     *
+     * @since 0.1
+     */
+    public function testValidateSuccess()
+    {
+        $subject = $this->createInstance();
+
+        $subject->validate(true);
+        $this->assertTrue(true, 'This line cannot be reached if validation fails.');
+    }
+
+    /**
+     * Tests whether invalid values do not pass validation.
+     *
+     * @expectedException \Dhii\Validation\Exception\ValidationFailedException
```

Comment:
Agreed, after reading Bergmann's [post](https://thephp.cc/news/2016/02/questioning-phpunit-best-practices). Even though we can't use the `class` class constant, since it is only introduced in PHP 5.5, the other benefits look useful.